src/lib/storage/dictionary_segment.cpp

```diff
@@ -39,6 +39,7 @@ AllTypeVariant DictionarySegment<T>::operator[](const ChunkOffset chunk_offset)
 
 template <typename T>
 std::shared_ptr<const pmr_vector<T>> DictionarySegment<T>::dictionary() const {
+  access_counter[SegmentAccessCounter::AccessType::Dictionary] += _dictionary->size();
```

Comment:
Increasing the counter by dictionary->size() is not correct for all cases but probably okay at most times...